src/blob.jl

```diff
@@ -8,54 +61,99 @@ struct Blob{T}
 
     function Blob{T}(base::Ptr{Nothing}, offset::Int64, limit::Int64) where {T}
         @assert isbitstype(T)
-        new(base, offset, limit)
+        @boundscheck _bounds_check(base, offset, limit, self_size(T), T)
+        new{T}(base, offset, limit)
+    end
+end
+
+@noinline function _bounds_check(
```

Comment:
What's the rationale for making this `@noinline`?  Julia doesn't do it for `checkbounds(::AbstractArray, I...)`, for example.

Reply:
The reason for the large amount of generate llvm code was the bounds check (building the strings etc). Therefore, I want to avoid that this is getting inlined into the specialized function.